integration-tests/src/test/java/com/datastax/oss/driver/mapper/DaoAnnotationTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.mapper;
+
+import static com.datastax.oss.driver.mapper.MapperProcessorTestUtil.loadLinesFromSourceFile;
+import static com.google.testing.compile.CompilationSubject.assertThat;
+import static com.google.testing.compile.Compiler.javac;
+
+import com.datastax.oss.driver.internal.mapper.processor.MapperProcessor;
+import com.google.testing.compile.Compilation;
+import com.google.testing.compile.JavaFileObjects;
+import java.io.IOException;
+import javax.tools.StandardLocation;
+import org.junit.Test;
+
+public class DaoAnnotationTest {
+
+  private static final String PACKAGE_NAME = "com.datastax.oss.driver.mapper.model.inventory";
+
+  @Test
+  public void should_generate_product_dao_impl() throws IOException {
+    // given
+    String sourceFileLocation =
+        "src/test/java/com/datastax/oss/driver/mapper/model/inventory/ProductDao.java";
+    String generatedSourceFileLocation =
+        this.getClass().getResource("/annotation-processor/ProductDao_Impl_Expected").getPath();
+
+    // when
+    Compilation compilation =
+        javac()
+            .withProcessors(new MapperProcessor())
+            .compile(
+                JavaFileObjects.forSourceLines(
+                    PACKAGE_NAME + ".ProductDao", loadLinesFromSourceFile(sourceFileLocation)));
+
+    // then
+    assertThat(compilation).succeededWithoutWarnings();
+    assertThat(compilation)
+        .generatedFile(StandardLocation.SOURCE_OUTPUT, PACKAGE_NAME, "ProductDao_Impl.java")
+        .hasSourceEquivalentTo(
+            JavaFileObjects.forSourceLines(
+                PACKAGE_NAME + "ProductDao_Impl_Expected",
```

Comment:
My only concern here is that this will create some maintenance overhead, as we add/remove or change our code generation.... I can't however come up with a better way to do this, so for now I think it will work

Reply:
I agree that it has some maintenance cost. 
But also, testing normal code (not annotation processor generation) in a unit test way and writing test cases for new functionalities has that cost. 
The generated code should be tested in some way and every change to generated class is something that should be reflected in the test. So I think that this maintenance cost is worth it.

Reply:
The problem is doing character-to-character comparison of the generated code. This is very sensitive to changes; if we rely on this too much, my fear is that we'll soon have to update a gazillion reference files any time we touch something. You've already run into that (last commit "update generated files"), and I'm seeing it again as I've pushed new changes to `java2078` and am rebasing this PR.

I think we should reserve this approach for:
* processor errors and warnings
* checking that files have been generated, without necessarily testing their contents (a great use case for that would be JAVA-2136: Support processing of nested types)

Also, the test code should be simple enough that it can be inlined directly in the test. That's more readable than having to open separate resource files. If something is too complicated to be inlined, it probably warrants a full-fledged integration test.

I'll push a few examples in that spirit once I'm done rebasing.

Reply:
> You've already run into that (last commit "update generated files")
My last commit was because on the `java2078` branch there was a new feature and I was rebasing this branch so I've added tests cases to cover those new functionalities. If we will add them when implementing this maybe it won't be painful.

>processor errors and warnings 

+1 on that. It is a good tool to validate if we are failing with the proper message

>checking that files have been generated, without necessarily testing their contents (a great use case for that would be JAVA-2136: Support processing of nested types)

It sounds good but I am a bit skeptical about that. Validating only the fact that file was generated properly sometimes may not be enough. But I am also +1 on doing this inline in code and maybe validate that specific line of code appeared in the output instead of validating whole content.